SoObjects/SOGo/WORequest+SOGo.m

```diff
@@ -180,14 +180,24 @@ - (BOOL) isMacOSXAddressBookApp
   cc = [self clientCapabilities];
 
   b = (
-       [[cc userAgent] rangeOfString: @"CFNetwork"].location != NSNotFound
-       && [[cc userAgent] rangeOfString: @"Darwin"].location != NSNotFound
-       || (
-           [[cc userAgent] rangeOfString: @"CFNetwork"].location != NSNotFound
-            || [[cc userAgent] rangeOfString: @"Mac OS X"].location != NSNotFound
-           )
-       && [[cc userAgent] rangeOfString: @"AddressBook"].location != NSNotFound
-       );
+        (
+          (
+            [[cc userAgent] rangeOfString: @"CFNetwork"].location != NSNotFound
+            && [[cc userAgent] rangeOfString: @"Darwin"].location != NSNotFound
+          )
+          ||
+          (
+            [[cc userAgent] rangeOfString: @"CFNetwork"].location != NSNotFound
+            &&
+            (
+              [[cc userAgent] rangeOfString: @"Mac OS X"].location != NSNotFound
+              ||
+              [[cc userAgent] rangeOfString: @"Mac_OS_X"].location != NSNotFound
+            )
+          )
+        )
+        && [[cc userAgent] rangeOfString: @"AddressBook"].location != NSNotFound
```

Comment:
The address book app on OSX 10.6 uses a localized user-agent string so we can't expect to locate **AddressBook**. Even the English localization will use **Address%20Book** as shown in the comments above the method.